recipes/gdptools/meta.yaml

```diff
@@ -0,0 +1,77 @@
+{% set name = "gdptools" %}
+{% set version = "0.0.15.dev0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/gdptools-{{ version }}.tar.gz
+  sha256: c21cae1bbd2edb69dbe815078c07fc8e74fde0f0608d6efa12cdc52cdbdb251e
+
+build:
+  entry_points:
+    - gdptools = gdptools.__main__:main
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  host:
+    - python >=3.6,<3.10
+    - pip
+    - poetry-core >=1.0.0
+  run:
+    - python >=3.6,<3.10
+    - python-attr >=0.3.2,<0.4.0
+    # - geopandas ==0.11.1
+    - geopandas
+    - netcdf4 >=1.5.8,<2.0.0
+    - numpy >=1.22.4,<2.0.0
+    - cftime >=1.6.0,<2.0.0
+    - click <8
+    - pydap >=3.2.2,<4.0.0
+    - pandas >=1.4.2,<2.0.0
+    - scipy >=1.8.1,<2.0.0
+    # - shapely ==1.8.4
+    - shapely
+    - pystac >=1.4.0,<2.0.0
+    # - pyproj >=3.3.1,<4.0.0
+    - pyproj
+    - zarr >=2.11.3,<3.0.0
+    - pygeoapi >=0.11.0,<0.12.0
+    - rioxarray >=0.11.1,<0.12.0
+    - rasterstats >=0.16.0,<0.17.0
+    - intake >=0.6.5,<0.7.0
+    - rasterio >=1.2.10,<2.0.0
+    - metpy >=1.3.0,<2.0.0
+    - dask-core >=2022.6.0,<2023.0.0
+    - xarray >=2022.3.0,<2023.0.0
+    # - pygeos ==0.12.0
+    - pygeos
+    - requests >=2.28.0,<3.0.0
+
+test:
+  files:
+   - tests/
+  requires:
+    - pip
+    - pytest
+  imports:
+    - gdptools
+  commands:
+    # Some of the pins are too aggressive making the package impossible to install.
```

Comment:
The pins are terribly tight. Would it be ok to remove some of the upper bounds? I guess you really already have on several of them.

I've been advising people to not have upper bounds unless they really know they need them. Maybe I'm wrong on that. I feel like here at conda-forge, we have repodata patching to fix/impose upper bounds when needed, and we should not prematurely introduce conflicts that don't really exist.

Reply:
That is why I added the tests. I relaxed all those that I could and ensure a passing build. The others will need some upstream modifications.

Yeah, folks usually confuse apps with libraries.